### PR TITLE
Build oneAPI DPC++ part on Windows

### PR DESCRIPTION
**Pull request by rlnx**
- Added support for DPC++ build on windows in makefile
- Updated DLL resource files
- Fix unresolved externals
- Add makefile_win for DPC++ examples

### CODE REVIEW

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

**Comment by napetrov** [comment]
/intelci: restart
set infra_branch=dev/risrafil-win-dpc

**Comment by napetrov** [comment]
/intelci: restart
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: restart
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: start
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

**Comment by rlnx** [comment]
/intelci: run
set infra_branch=dev/risrafil-win-dpc

File: cpp/daal/include/services/internal/dynamic_lib_helper.h [hunk 1]
```diff
@@ -94,7 +94,7 @@ class DynamicLibHelper final
             return NULL;
         }
     #elif defined(_WIN32) || defined(_WIN64)
-        void * sym = GetProcAddress((HMODULE)_handle, symName);
+        void * sym = (void *)GetProcAddress((HMODULE)_handle, symName);
```
Comment (michael-smirnov) [inline cpp/daal/include/services/internal/dynamic_lib_helper.h:97]: minor: it's better to use reinterpret_cast here

**Review by michael-smirnov — APPROVED** [review]

**Comment by michael-smirnov** [comment]
@rlnx, do we plan to include this in Gold as well?

**Pull request merged**